File: php/features.php
<?php
declare(strict_types=1);

// TempMailApiByBoomlify SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/RatelimitFeature.php';
require_once __DIR__ . '/feature/RetryFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';
require_once __DIR__ . '/feature/TimeoutFeature.php';


class TempMailApiByBoomlifyFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new TempMailApiByBoomlifyBaseFeature();
            case "ratelimit":
                return new TempMailApiByBoomlifyRatelimitFeature();
            case "retry":
                return new TempMailApiByBoomlifyRetryFeature();
            case "test":
                return new TempMailApiByBoomlifyTestFeature();
            case "timeout":
                return new TempMailApiByBoomlifyTimeoutFeature();
            default:
                return new TempMailApiByBoomlifyBaseFeature();
        }
    }

    /**
     * Does a generated feature class back this name? False for a name only
     * an options extend instance can supply (the station adopt path) - the
     * constructor uses this to skip make_feature for such names instead of
     * adding a stray BaseFeature.
     */
    public static function has_feature(string $name): bool
    {
        switch ($name) {
            case "base":
            case "ratelimit":
            case "retry":
            case "test":
            case "timeout":
                return true;
            default:
                return false;
        }
    }
}

File: py/tempmailapibyboomlify_sdk/features.py
# TempMailApiByBoomlify SDK feature factory

from tempmailapibyboomlify_sdk.feature.base_feature import TempMailApiByBoomlifyBaseFeature
from tempmailapibyboomlify_sdk.feature.ratelimit_feature import TempMailApiByBoomlifyRatelimitFeature
from tempmailapibyboomlify_sdk.feature.retry_feature import TempMailApiByBoomlifyRetryFeature
from tempmailapibyboomlify_sdk.feature.test_feature import TempMailApiByBoomlifyTestFeature
from tempmailapibyboomlify_sdk.feature.timeout_feature import TempMailApiByBoomlifyTimeoutFeature


_FEATURES = {
    "base": lambda: TempMailApiByBoomlifyBaseFeature(),
    "ratelimit": lambda: TempMailApiByBoomlifyRatelimitFeature(),
    "retry": lambda: TempMailApiByBoomlifyRetryFeature(),
    "test": lambda: TempMailApiByBoomlifyTestFeature(),
    "timeout": lambda: TempMailApiByBoomlifyTimeoutFeature(),
}


def _make_feature(name):
    factory = _FEATURES.get(name)
    if factory is not None:
        return factory()
    return _FEATURES["base"]()


# True when this SDK was generated with the named feature class - the
# constructor's tolerance for extend-carried features reads this (an
# active name with no generated class must not become a BaseFeature
# stray when an extend instance carries it).
def _has_feature(name):
    return name in _FEATURES
